### PR TITLE
Add unread messages count

### PR DESCRIPTION
**Pull request by JorgeMichelena**
-Modified Match serializer to include new fields showing the content of last message sent in the conversation and the amount of unreads messages by the user.
-Modified ChatConsumer, now when a user enters a chatroom all messages marked as unread by that user are marked as read.
-Created test cases

### CODE REVIEW

File: chat/tests/test_unread_messages.py [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+    def setUp(self):
+        super().setUpClass()
+        self.client = Client()
+        #define data to make users and then register them
+        data1 = { 
+            'username': 'usuario1',
+            'email': 'usuario1@example.com',
+            'password1': 'test1234password',
+            'password2': 'test1234password',
+            'gender': User.MALE
```
Comment (kaozdl) [inline chat/tests/test_unread_messages.py:28]: why are these not faked?

File: chat/tests/test_unread_messages.py [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+        }
+        data2 = { 
+            'username': 'usuario2',
+            'email': 'usuario2@example.com',
+            'password1': 'test1234password',
+            'password2': 'test1234password',
+            'gender': User.MALE
```
Comment (kaozdl) [inline chat/tests/test_unread_messages.py:35]: same here, also try to code in english

File: api/serializers.py [hunk 3]
```diff
@@ -49,7 +49,19 @@ def create(self, validated_data):
...
+        fields = ['pk', 'creation_date', 'unread_messages', 'last_message', 'target1', 'target2']
+
+    def get_unread_messages(self, instance):
+        user_id = self.context['request'].user.id
+        return instance.chatlog.filter(date_seen__isnull=True).exclude(author__id=user_id).count()
+
+    def get_last_message(self, instance):
+        last_msg = ''
+        if instance.chatlog.filter(date_seen__isnull=True).count() != 0:
+            last_msg = instance.chatlog.filter(date_seen__isnull=True).order_by('id').reverse()[0].content
```
Comment (brunomichetti) [inline api/serializers.py:66]: Here you can do something like this:
```
instance_queryset = instance.chatlog.filter(date_seen__isnull=True)
if instance_queryset:
    last_msg = instance_queryset.order_by('-id')[0].content
```
To improve efficiency.

File: chat/consumers.py [hunk 4]
```diff
@@ -5,36 +5,43 @@
...
+        self.connected[self.user.id] = connections + 1
         self.accept()
 
     def disconnect(self, close_code):
         async_to_sync(self.channel_layer.group_discard)(
             self.room_group_name,
             self.channel_name
         )
-        self.connected.discard(self.user.id)
+        self.connected[self.user.id] = self.connected[self.user.id] - 1
```
Comment (brunomichetti) [inline chat/consumers.py:34]: Why do you do ```self.connected[self.user.id] - 1```?

Comment (JorgeMichelena) [reply chat/consumers.py:34]: The `connected` dictionary is meant to keep track of the number of websocket connections every user has, having the user id as key and the number of connections as value. That's why the connect method of the consumer increments by 1 the value associated to the user and the disconnect method decrements that value also by 1.

**Pull request merged**